stan/math/prim/mat/prob/ordered_logistic_glm_lpmf.hpp

```diff
@@ -0,0 +1,174 @@
+#ifndef STAN_MATH_PRIM_MAT_PROB_ORDERED_LOGISTIC_GLM_LPMF_HPP
+#define STAN_MATH_PRIM_MAT_PROB_ORDERED_LOGISTIC_GLM_LPMF_HPP
+
+#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
+#include <stan/math/prim/scal/meta/partials_return_type.hpp>
+#include <stan/math/prim/scal/meta/operands_and_partials.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
+#include <stan/math/prim/scal/err/check_bounded.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/mat/fun/value_of_rec.hpp>
+#include <stan/math/prim/arr/fun/value_of_rec.hpp>
+#include <stan/math/prim/mat/meta/as_column_vector_or_scalar.hpp>
+#include <stan/math/prim/scal/meta/as_column_vector_or_scalar.hpp>
+#include <stan/math/prim/scal/meta/include_summand.hpp>
+#include <stan/math/prim/scal/fun/size_zero.hpp>
+#include <stan/math/prim/mat/err/check_ordered.hpp>
+#include <stan/math/prim/arr/err/check_ordered.hpp>
+#include <stan/math/prim/mat/fun/log1m_exp.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the log PMF of the ordinal regression Generalized Linear Model (GLM).
+ * This is equivalent to and faster than ordered_logistic_lpmf(y, x * beta,
+ * cuts).
+ *
+ * @tparam T_y type of integer vector of dependent variables (labels)
+ * @tparam T_x_scalar type of elements in the matrix of independent variables
+ * (features)
+ * @tparam T_beta T_beta type of the weight vector
+ * @tparam T_cuts T_beta type of the cutpoints vector
+ * @param y integer vector parameter
+ * @param x design matrix
+ * @param beta weight vector
+ * @param cuts vector of cutpoints
```

Comment:
I assume this only accepts a column vector, since that's what we take "vector" to mean. Otherwise, please indicate this can be a row vector or column vector.  We want the doc to be clear on types here since the templating isn't providing that information when you template the whole container.

I'm just adding single comments rather than reviewing as I do'nt want to overreview someone else who might be doing this.